presto-orc/src/main/java/com/facebook/presto/orc/writer/ColumnWriters.java

```diff
@@ -139,6 +143,39 @@ public static ColumnWriter createColumnWriter(
                         integerDictionaryEncodingEnabled);
                 return new MapColumnWriter(columnIndex, compressionParameters, dwrfEncryptor, orcEncoding, keyWriter, valueWriter, metadataWriter);
             }
+            */
+            case FLATMAP: {
+                int keyIndex = orcType.getFieldTypeIndex(0);
+                OrcType keyType = orcTypes.get(keyIndex);
+                ColumnWriterFactory valueWriterfactory =
+                    new ColumnWriterFactory()
+                        .setColumnIndex(orcType.getFieldTypeIndex(1))
+                        .setOrcTypes(orcTypes)
+                        .setType(type.getTypeParameters().get(1))
+                        .setCompressionParameters(compressionParameters)
+                        .setOrcEncoding(orcEncoding)
+                        .setHiveStorageTimeZone(hiveStorageTimeZone)
+                        .setStringStatisticsLimit(stringStatisticsLimit)
+                        .setDwrfEncryptors(dwrfEncryptors)
+                        .setMetadataWriter(metadataWriter)
+                        .setIntegerDictionaryEncodingEnabled(integerDictionaryEncodingEnabled);
+                switch (keyType.getOrcTypeKind()) {
+                    case CHAR:
+                        checkArgument(orcEncoding != DWRF, "DWRF does not support %s type", type);
```

Comment:
FlatMap encoding is only supported by DWRF format. ORC format, does not support flat map encoding.